docs/reference/ccr/apis/auto-follow/put-auto-follow-pattern.asciidoc

```diff
@@ -0,0 +1,113 @@
+[role="xpack"]
+[testenv="platinum"]
+[[ccr-put-auto-follow-pattern]]
+=== Create Auto-Follow Pattern API
+++++
+<titleabbrev>Create Auto-Follow Pattern</titleabbrev>
+++++
+
+Creates an auto-follow pattern.
+
+==== Description
+
+This API creates a new named collection of auto-follow patterns against the
+remote cluster specified in the request body. Newly created indices on the
+remote cluster matching any of the specified patterns will be automatically
+configured as follower indices.
+
+==== Request
+
+[source,js]
+--------------------------------------------------
+PUT /_ccr/auto_follow/<auto_follow_pattern_name>
+{
+  "leader_cluster" : "<leader_cluster>",
+  "leader_index_patterns" :
+  [
+    "<leader_index_pattern>"
+  ],
+  "follow_index_pattern" : "<follow_index_pattern>"
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[setup:leader_cluster]
+// TEST[s/<auto_follow_pattern_name>/auto_follow_pattern_name/]
+// TEST[s/<leader_cluster>/leader_cluster/]
+// TEST[s/<leader_index_patterns>/leader_index*/]
+// TEST[s/<follow_index_pattern>/{{leader_index}}-follower/]
+
+//////////////////////////
+
+[source,js]
+--------------------------------------------------
+DELETE /_ccr/auto_follow/auto_follow_pattern_name
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
+//////////////////////////
+
+Here, `<auto_follow_pattern_name>` is a user-chosen name for the collection of
+auto-follow patterns being created, `<leader_cluster>` is the
+<<modules-remote-clusters,remote cluster>> containing the leader indices to
+match against, `<leader_index_pattern>` is a simple index pattern to match
+against indices in the remote cluster, and `<follow_index_pattern>` is the
+pattern to use to name the resulting follower indices. The
+`<follow_index_pattern>` can use the template `{{leader_index}}` to refer to the
+name of the leader index being followed.
+
```

Comment:
==== Path Parameters

`auto_follow_pattern_name` (required)::
  (string) Identifier for the auto-follow patterns.